opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -99,6 +101,30 @@ public function index() {
         $this->data['rss_feed_title'] = $this->language->get('rss_feed_title');
         $this->data['rss_feed_text'] = $this->language->get('rss_feed_text');
         $this->data['rss_feed_url'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss';
+
+        // Layout text and buttons
+        $this->data['entry_title_layout'] = $this->language->get('entry_title_layout');
+        $this->data['entry_title_position'] = $this->language->get('entry_title_position');
+        $this->data['entry_title_status'] = $this->language->get('entry_title_status');
+        $this->data['entry_title_sort_order'] = $this->language->get('entry_title_sort_order');
+        $this->data['entry_select_top'] = $this->language->get('entry_select_top');
+        $this->data['entry_select_bottom'] = $this->language->get('entry_select_bottom');
+        $this->data['entry_select_left'] = $this->language->get('entry_select_left');
+        $this->data['entry_select_right'] = $this->language->get('entry_select_right');
+        $this->data['entry_button_module'] = $this->language->get('entry_button_module');
+        $this->data['button_remove'] = $this->language->get('button_remove');
+
+        // Layout module
+        $this->load->model('design/layout');
+        $this->data['layouts'] = $this->model_design_layout->getLayouts();
+        $this->data['modules'] = array();
+        // Fetch modules
+        if (isset($this->request->post['smaily_for_opencart_module']) && is_array($this->request->post['smaily_for_opencart_module'])) {
+            $this->data['modules'] = $this->request->post['smaily_for_opencart_module'];
+        } elseif ($this->config->get('smaily_for_opencart_module')) {
+            $this->data['modules'] = $this->config->get('smaily_for_opencart_module');
```

Comment:
What happens if `smaily_for_opencart_module` setting does not exist?

Reply:
$this->data['modules'] will be null

Reply:
I can see that, but what happens if `$this->data['modules']` is empty :)

Reply:
I ran var_dump, the `data['modules'] will be an empty array to be precise. The foreach that uses `data['modules']` wont start and the settings for the opt-in form will not be shown.
I'm not quite sure what you're getting at.

Reply:
If I understand correctly `$this->config->get('smaily_for_opencart_module')` returns `NULL` if `smaily_for_opencart_module` setting is not found, and it shouldn't exist when you have not configured Smaily of OpenCart module. Am I correct?

Reply:
You are right, yes.

Reply:
AAh. Ok, when you remove something the array is empty, and you can't save it. Right?

Reply:
I think I may have understood that code part wrong then.
But doesn't `elseif ($this->config->get('smaily_for_opencart_module'))`  
check `TRUE` if it exists and `FALSE` if it gets something else (like null), in which case it won't pass along NULL to the template?

Reply:
Oh, my bad. I somehow missed that row.

Although. You should still check if `$this->config->get('smaily_for_opencart_module')` returns something you expect, like an array. If for whatever reason (bug, corrupted setting value) returned values should be some other truthy value then it would still break the template.

Rule of thumb with handling data is - always be strict (or explicit) on what type of data you expect, anything else you reject.